common/lib/xmodule/xmodule/tests/test_vertical.py

```diff
@@ -202,14 +213,31 @@ def test_render_access_denied_blocks(self, has_access_error):
         self.module_system._services['user'] = StubUserService()
         self.vertical.due = datetime.now(pytz.UTC) + timedelta(days=-1)
         self.problem_block.has_access_error = has_access_error
+        self.nested_problem_block.has_access_error = has_access_error
 
         context = {'username': self.username, 'hide_access_error_blocks': True}
         html = self.module_system.render(self.vertical, STUDENT_VIEW, context).content
 
         if has_access_error:
             assert self.test_problem not in html
+            assert self.test_problem_nested not in html
         else:
             assert self.test_problem in html
+            assert self.test_problem_nested in html
+
+    @ddt.data(True, False)
+    def test_block_has_access_error(self, has_access_error):
+        """ Tests block_has_access_error gives the correct result for child node questions """
+        # Use special block from setup (vertical/nested_vertical/problem)
+        # has_access_error is set on problem an extra level down, so we have to recurse to pass
+
+        self.nested_problem_block.has_access_error = has_access_error
+        should_block = self.vertical.block_has_access_error(self.vertical)
+
+        if has_access_error:
+            assert should_block
+        else:
+            assert not should_block
```

Comment:
Nit: if you want to reduce lines, you could just do `assert should_block == has_access_error`

Reply:
ah of course! thanks!